src/test/java/com/rultor/agents/github/qtn/QnReleaseTest.java

```diff
@@ -77,4 +79,48 @@ public void buildsRequest() throws Exception {
         );
     }
 
+    /**
+     * QnRelease can deny release when tag is outdated.
+     * @throws Exception In case of error.
+     */
+    @Test
+    public void denyOutdatedTag() throws Exception {
+        final Repo repo = new MkGithub().randomRepo();
+        final Issue issue = repo.issues().create("", "");
+        repo.releases().create("1.7");
+        issue.comments().post("release `1.6`");
+        MatcherAssert.assertThat(
+            new QnRelease().understand(
+                new Comment.Smart(issue.comments().get(1)), new URI("#")
+            ),
+            Matchers.is(Req.EMPTY)
+        );
+        MatcherAssert.assertThat(
+            new Comment.Smart(issue.comments().get(2)).body(),
+            Matchers.containsString("There is already a release `1.7`")
+        );
+        issue.comments().post("release");
+    }
+
+    /**
+     * QnRelease can deny release when tag name is not given.
```

Comment:
@mkordas I also see your point :), but following that rule strictly we would simply keep the old test and adjust the code to simply handle the absence of a version equal to a valid version tag ?
It seems to me we're sort of acting against the gist of making an understandable fix here, by adding incorrect complication ( and documenting it ) instead of adding less but correct complication.
Let me handle the other things tonight, maybe I'll find a better approach or argument since we were gonna continue tomorrow on this anyways, but you're right for now I think.